crates/wasm-mutate/src/mutators/peephole/dfg/mod.rs

```diff
@@ -0,0 +1,687 @@
+use std::{collections::HashMap, hash::Hash};
+
+use wasmparser::{Operator, Range};
+
+use super::TupleType;
+
+/// It executes a minimal symbolic evaluation of the stack to detect operands location in the code for certain operators
+/// For example, i.add operator should know who are its operands
+pub struct DFGIcator {}
+
+#[derive(Debug)]
+pub struct BBlock {
+    pub(crate) range: Range,
+}
+
+#[derive(Debug, Clone)]
+pub struct StackEntry {
+    pub operator_idx: usize,
+    pub data: Box<StackEntryData>,
+    pub byte_stream_range: Range,
+}
+
+#[derive(Debug, Clone)]
+pub enum StackEntryData {
+    Leaf,
+    Node { operands: Vec<usize> },
+    Unknown,
+}
+
+#[derive(Debug)]
+pub struct MiniDFG {
+    // Some of the operators have no stack entry
+    // This will help to decide or not to mutate the operators, avoiding egrapphp creation, etc
+    pub map: HashMap<usize, usize>,
+    // Each stack entry represents a position in the operators stream
+    // containing its children
+    pub entries: Vec<StackEntry>,
+    // For each stack entry we keep the parental relation, this will help to write down the dfg
+    // to Wasm
+    // We write each stack entry having no parent, i.e. a root in the dfg
+    pub parents: Vec<i32>,
+}
+
+impl<'a> DFGIcator {
+    pub fn new() -> Self {
+        DFGIcator {}
+    }
+
+    /// Linear algorithm  to detect basic blocks
+    /// This follows the tradition way to detect them
+    /// 1 - Every jump instruction creates a new BB right after (in wasm: br, br_if, loop, block, if, else)
+    /// 2 - Every operator that could be a target of a jump also starts a BB (end ... :) Wasm always jumps to end)
+    /// 3 - The first operator is the start of a BB
+    pub fn collect_bbs(
+        &self,
+        operators: &[TupleType],
+        onbb: Option<&dyn Fn(&BBlock)>,
+    ) -> crate::Result<Vec<BBlock>> {
```

Comment:
Actually, we dont need this methid enymore. This was used to collect all basic blocks from the function. Since we only need to collect the current one, I will remove it